### DISCUSSION TITLE
Build failing for `aarch64-unknown-linx-musl`

### DISCUSSION THREAD

**Discussion by littlebenlittle** [cross-rs/cross#578] [Q&A] [answered]
I'm trying to build a simple [wry](https://github.com/tauri-apps/wry) app for `aarch64-unknown-linx-musl`.

[branch with full example](https://github.com/littlebenlittle/wry-aarch64-musl-example/tree/cross)

My container build script

```bash
ctr=`buildah from rustembedded/cross:aarch64-unknown-linux-musl-0.2.1`
buildah run $ctr apt-get update
buildah run $ctr apt-get install -y \
	libwebkit2gtk-4.0-dev \
	libgtksourceviewmm-3.0-dev \
	libc6-dev-i386 \
	clang
buildah commit $ctr cross-wry:aarch64-unknown-linux-musl-0.2.
```

My `Cross.toml`

```toml
[target.aarch64-unknown-linux-musl]
image="cross-wry:aarch64-unknown-linux-musl-0.2.1"
```

This results in

```
   Compiling test v0.1.0 (/project)
error: linking with `aarch64-linux-musl-gcc` failed: exit status: 1
...
  = note: /usr/local/bin/../lib/gcc/aarch64-linux-musl/6.4.0/../../../../aarch64-linux-musl/bin/ld: cannot find -lwebkit2gtk-4.0
          /usr/local/bin/../lib/gcc/aarch64-linux-musl/6.4.0/../../../../aarch64-linux-musl/bin/ld: cannot find -lgtk-3
          /usr/local/bin/../lib/gcc/aarch64-linux-musl/6.4.0/../../../../aarch64-linux-musl/bin/ld: cannot find -lgdk-3
...
          /usr/local/bin/../lib/gcc/aarch64-linux-musl/6.4.0/../../../../aarch64-linux-musl/bin/ld: cannot find -lgobject-2.0
          /usr/local/bin/../lib/gcc/aarch64-linux-musl/6.4.0/../../../../aarch64-linux-musl/bin/ld: cannot find -lglib-2.0
          collect2: error: ld returned 1 exit status
```

I'm very new to linking so I'm not really sure where to start debugging. My best guess here is that packages installed to the container aren't appropriate for this build, perhaps because of architecture.

This is related to [this issue](https://github.com/tauri-apps/wry/issues/357).

**Comment by Alexhuszagh** [comment, marked as answer]
Sorry for the late response. There's a few issues here: one is that you're using packages linked against `glibc` for a `musl` build. This may work fine during compilation, but the binary is unlikely to run. Next, you're using much older versions of our images: v0.2.1. Try using `ghcr.io/cross-rs/aarch64-unknown-linux-musl:0.2.4` (latest release) or `ghcr.io/cross-rs/aarch64-unknown-linux-musl:main` (latest development branch, which uses a newer Ubuntu base). However, it will likely be a better idea to use an Alpine-based image for this purpose. I've been meaning to add an `x86_64-unknown-linux-musl` Alpine image just like how we support a CentOS one for `x86_64-unknown-linux-gnu`. You will likely have to build GTK from source otherwise.
